File: Prácticas/P1/test/src/Weapon.java
/**
 * @author elena
 * @brief Esta clase representa a las armas de las que puede disponer una estación 
 * espacial para potenciar su energía al disparar.
 */
class Weapon {
    private String name;
    private WeaponType type;
    private int uses;
    
    //Constructores
    Weapon(String n, WeaponType t, int u){
        this.name= n;
        this.type= t;
        this.uses= u;
    }
    
    Weapon(Weapon w){
        this(w.name, w.getType(), w.getUses());
    }
    
    //Consultores 
    public WeaponType getType(){
        return type;
    }
    
    public int getUses(){
        return uses;
    }
    
    /*
     * @brief Devuelve la potencia de disparo indicada por el tipo de arma.
    */
    public float power(){
        return type.getPower();
    }
    
    /*
    * @brief Si el valor del atributo uses es mayor que 0, lo decrementa en una 
    * unidad y devuelve el valor del método power(); devuelve el valor 1.0 en 
    * otro caso.
    */
    public float useIt(){
        if( uses > 0){
            uses--;
            return power();
        }
        return 1.0f;
    }
}

File: Prácticas/ruby/lib/weapon.rb
=begin
Representa a las armas de las que puede disponer una estación espacial para potenciar su
energía al disparar
=end

require_relative 'WeaponToUI'

module Deepspace
  class Weapon
    def initialize(name, type, uses)
      @name= name
      @type= type
      @uses= uses
    end

    def newCopy(w) #Aqui deberias subir self
      Weapon.new(w.name, w.type, w.uses)
    end

    attr_reader :name, :type, :uses

    def power
      @type.power
    end

    def useIt
      if @uses > 0
        @uses -= 1
        power
      else
        1.0
      end
    end
    
    def getUIversion
      WeaponToUI.new(self)
    end
  end
end
